domain/src/main/java/today/stepbeyond/examples/springbootexamples/domain/usecases/PetRegistrationUseCases.java

```diff
@@ -0,0 +1,32 @@
+package today.stepbeyond.examples.springbootexamples.domain.usecases;
+
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+import today.stepbeyond.examples.springbootexamples.domain.core.DogDomainService;
+import today.stepbeyond.examples.springbootexamples.domain.core.model.Dog;
+import today.stepbeyond.examples.springbootexamples.domain.infrastructure.controller.PetRegistrationInputPort;
+import today.stepbeyond.examples.springbootexamples.domain.infrastructure.repository.PetRepository;
+
+@Service
+public class PetRegistrationUseCases implements PetRegistrationInputPort {
```

Comment:
NIT: We should document why we chose that for simplicity. 

PetRegistrationInputPortService using one or more "use cases" to unregister a pet would be more flexible for adding business logic to the process. e.g. unregistration also triggers a new use case like "CleanUpCage" use case etc.